### ISSUE TITLE
TypeError exception raised

### ISSUE THREAD

**Issue by mwald84** [DIGITALCRIMINAL/OnlyFans#313]
Hi,

I recently updated to version 5 (5.1) of OnlyFans Scrapper and I get this error when running it:

2020-07-02 15:28:24,796 ERROR errors 'NoneType' object is not subscriptable
Traceback (most recent call last):
  File "D:\OneDrive\cb\grab_of5\datascraper\main_datascraper.py", line 114, in start_datascraper
    session["session"], app_token, session["subscriber_count"], me_api, auth_count)
  File "D:\OneDrive\cb\grab_of5\modules\onlyfans.py", line 779, in get_subscriptions
    results.sort(key=lambda x: x["subscribedByData"]['expiredAt'])
  File "D:\OneDrive\cb\grab_of5\modules\onlyfans.py", line 779, in <lambda>
    results.sort(key=lambda x: x["subscribedByData"]['expiredAt'])
TypeError: 'NoneType' object is not subscriptable

what am I may be doing wrong? I followed the config setup instructions as in the main page (similar to what it used to be on previous versions of the app).

There's accounts I'm not subbed anymore on OnlyFans (it seemed relevant mentioning it because of the key it's trying to access when the error raises:  x["subscribedByData"]['expiredAt']). Previous versions of the app had a config option to avoid following them ("ignore_unfollowed_accounts": "all") but it's not present anymore. I wonder if this is related.

I'm using Python 3.7.4 on Windows 10

Thanks in advance

Marcos

**Comment by mwald84** [comment]
Update:

I tried to add a print(x) before the line with the error to try getting more info about the problem and I got this:

Traceback (most recent call last):
  File "D:\OneDrive\cb\grab_of5\datascraper\main_datascraper.py", line 114, in start_datascraper
    session["session"], app_token, session["subscriber_count"], me_api, auth_count)
  File "D:\OneDrive\cb\grab_of5\modules\onlyfans.py", line 779, in get_subscriptions
    print(x)
UnboundLocalError: local variable 'x' referenced before assignment

so it seems the var is not set for some reason.

**Comment by mwald84** [comment]
I added a print(results) before the line raising the exception and the data is there for othre key. Specifically for results["subscribedByData"] it has None stored (as the exception suggests).

I also realized that the current data in 'results' is from my account (I'm a creator too). This may be the reason.

**Comment by mwald84** [comment]
I'm trying to track the problem to help.

I inserted a breakpoint() at line 750 from onlyfans.py, just before 

    results = pool.starmap(multi, product(
        offset_array, [session]))

and tried to inspect to give more insights about the problem.

the resulting list after that line is executed has a list with the following content:
results[0] is my username (I'm a creator too so this may be the issue) and my data
results[1] have a list with the 4 users I follow and their data

after executing the next line:

results = list(chain(*results))

it returns results as a list with 5 positions.
At first position [0] is my data and ["subscribedByData"]['expiredAt'] (which raised the error) is absent
from [1] to [4] are the people I follow with all data (including ["subscribedByData"]['expiredAt'])

So I guess the problem is the fact I'm a creator. I tried to add myself to "blacklist_name" on conf.json but it didn't solve the problem.

I fixed the problem by changing the line 752 from:

    results = list(chain(*results))

to

    results = results[1]

I'm not sure if this is the best way to fix it. But it's working for me.
